pubsub/aws/snssqs/subscription_mgmt.go

```diff
@@ -0,0 +1,180 @@
+package snssqs
+
+import (
+	"context"
+	"sync"
+
+	"github.com/puzpuzpuz/xsync/v3"
+
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/kit/logger"
+)
+
+type (
+	SubscriptionAction int
+)
+
+const (
+	Subscribe SubscriptionAction = iota
+	Unsubscribe
+)
+
+var initOnce sync.Once
```

Comment:
I think this should be a property of the SubscriptionManger struct? Otherwise, if you have 2 components of the same type, one of them will not have the initOnce callback executed